### ISSUE TITLE
Type narrowing an enum shouldn't narrow to a literal

### ISSUE THREAD

**Issue by A5rocks** [python/mypy#19753]
**Bug Report**

If you compare an enum to a literal, then the type becomes that literal. Somewhat a followup to https://github.com/python/mypy/pull/19594.

**To Reproduce**

```python
from enum import Enum

class A(str, Enum):
    a = "b"
    b = "a"

def test(a: A) -> None:
    if a == "b":
        reveal_type(a)
```

**Expected Behavior**

```
main.py:9: note: Revealed type is "Literal[A.a]"
Success: no issues found in 1 source file
```

**Actual Behavior**

```
main.py:9: note: Revealed type is "Literal['b']"
Success: no issues found in 1 source file
```

**Your Environment**

Reproduced in mypy playground.
- Mypy version used: 1.17.1
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.12

**Comment by A5rocks** [comment]
Interestingly, if there's only 1 enum member, then the `reveal_type` becomes unreachable? No matter what the enum member is.

**Comment by sterliakov** [comment]
> if there's only 1 enum member

with #19594 it's handled better: still `Literal['b']` if `b` is among enum values, unreachable otherwise.

Also this problem does not exist for `StrEnum`, only `(str, Enum)` / `(int, Enum)` subclasses are affected.
